### PR TITLE
Clean up https://github.com/zendesk/rinku/pull/3

### PR DESCRIPTION
**Pull request by jcheatham**
Paired with @barker1977 to fix the test and get this prepped to look into fixing a punycode autolinking vulnerability.  Clean up includes:

 * jumping back a character after encountering a terminal character in check_domain so that the test pass / code works
 * redefining the unicode space function declaration to eliminate:
    `warning: implicit declaration of function 'isUnicodeSpace' is invalid in C99 [-Wimplicit-function-declaration]`
 * Renaming isUnicodeSpace to is_unicode_space to better fit in with existing code.

Considered reducing the strncmp check down to a collection of byte comparisons, since there's a bunch of redundancy in the unicode, but will leave that alone for now.

/cc @zendesk/sustaining 

### Risks
None

### CODE REVIEW

**Comment by jcheatham** [comment]
oh, also @shajith  of course

File: ext/rinku/autolink.c [hunk 1]
```diff
@@ -136,19 +139,19 @@ static size_t
...
-        while(link_end != 0) {
-          next_link_end = autolink_delim(data, link_end, offset, size);
-          if (next_link_end == link_end || iterations > 5) {
-            break;
-          }
-          link_end = next_link_end;
-          iterations++;
-        }
+	while(link_end != 0) {
+		next_link_end = autolink_delim(data, link_end, offset, size);
```
Comment (craig-day) [inline ext/rinku/autolink.c]: I was reading this and then all of a sudden... WHITESPACE. Why so much space?

Comment (jcheatham) [inline ext/rinku/autolink.c]: that's just the way things go sometimes.  Also, the file is riddled with tabs, while I'm betting @shajith had his editor set to use spaces... this brings it back into sync with the original style.  You can thank @barker1977

**Comment by barker1977** [comment]
:+1:

**Pull request merged**